### PR TITLE
Feature/paralle coord

### LINKED ISSUE #54: Link parallele coordinates to the backend
Add the link to the back end to populate the parallele coordinates

### PR DESCRIPTION
**Pull request by callicles**
Fixes #54 #57

### CODE REVIEW

File: app/models/services/RepositoryService.scala [hunk 1]
```diff
@@ -162,31 +157,26 @@ class RepositoryService @Inject()(
   }
 
   /**
-   * Check if the user has contributed to a given repo
-   * @param repoName repository user contributed or not
-   * @param user user that is being evaluated
-   * @return true if the user contributed and false if not.
+   * get all the scoring made for a repository for the given page and item per page.
```
Comment (jacobtardieu) [inline app/models/services/RepositoryService.scala]: Capital G.

File: app/models/services/RepositoryService.scala [hunk 2]
```diff
@@ -223,38 +214,54 @@ class RepositoryService @Inject()(
...
+                      user: User,
+                      repositoryName: String,
+                      scoreDocumentation: Int,
+                      scoreMaturity: Int,
+                      scoreDesign: Int,
+                      scoreSupport: Int,
+                      feedback: String): Future[Repository] = {
+
+    repoDAO.find(owner + "/" + repositoryName).flatMap({
+      case Some(repo) => Future.successful(scoreService.createScore(
```
Comment (jacobtardieu) [inline app/models/services/RepositoryService.scala]: Is the `createScore` method instantaneous (I mean no complicated computation or access to DBs)? If not do not use `.succesful`.

File: app/models/services/RepositoryService.scala [hunk 3]
```diff
@@ -223,38 +214,54 @@ class RepositoryService @Inject()(
...
+        user, repo, scoreDocumentation, scoreMaturity, scoreDesign, scoreSupport, feedback))
+      case None => oAuth2InfoDAO.find(user.loginInfo)
+        .flatMap(gitHub.getRepository(owner + "/" + repositoryName, _)
+          .flatMap(repo => repoDAO.create(repo.get))
+          .flatMap(repo => Future.successful(scoreService.createScore(
```
Comment (jacobtardieu) [inline app/models/services/RepositoryService.scala]: Same as above.

File: app/models/services/RepositoryService.scala [hunk 4]
```diff
@@ -223,38 +214,54 @@ class RepositoryService @Inject()(
...
     })
   }
 
-
   /**
    * Recalculate score for repo
+   *
    * @param repository repo to recalculate
    */
   def calculateScoreForRepo(repository: Repository): Future[(Int)] = {
```
Comment (jacobtardieu) [inline app/models/services/RepositoryService.scala]: No need for parenthesis around `Int`.

File: conf/routes [hunk 5]
```diff
@@ -3,13 +3,16 @@
...
+
+GET         /authenticate/:provider                        @controllers.SocialAuthController.authenticate(provider)
 
 
 # Map static resources from the /public folder to the /assets URL path
-GET        /assets/*file                              controllers.Assets.at(path="/public", file)
-GET        /webjars/*file                             controllers.WebJarAssets.at(file)
\ No newline at end of file
+GET         /assets/*file                                  controllers.Assets.at(path="/public", file)
+GET         /webjars/*file                                 controllers.WebJarAssets.at(file)
```
Comment (jacobtardieu) [inline conf/routes]: No newline :(

**Comment by jacobtardieu** [comment]
Quickly read but looks good!

File: app/models/services/RepositoryService.scala (hunk 2 above)
Comment (callicles) [inline app/models/services/RepositoryService.scala]: ? It is already `Future.successful`

File: app/models/services/RepositoryService.scala (hunk 3 above)
Comment (callicles) [inline app/models/services/RepositoryService.scala]: What's '.successful' I got an error when I try to delete the Future

File: public/javascripts/repository.js [hunk 6]
```diff
@@ -2,142 +2,168 @@
...
+    }
+
+    var svg = d3.select("#scoreChart").append("svg")
+        .attr("width", width + margin.left + margin.right)
+        .attr("height", height + margin.top + margin.bottom)
+        .append("g")
+        .attr("transform", "translate(" + margin.left + "," + margin.top + ")");
+
+    // Extract the list of dimensions and create a scale for each.
+    x.domain(dimensions = d3.keys(data[0]).filter(function(d) {
```
Comment (callicles) [inline public/javascripts/repository.js]: Codacy Issue found: [Variable 'dimensions' implicitly declared.](https://www.codacy.com/app/gitlinks/gitrank-web/file/993762770/issues/source?bid=2432310&fileBranchId=2868924#l62)

File: public/javascripts/repository.js [hunk 7]
```diff
@@ -2,142 +2,168 @@
...
+        return d != "timestamp" && (y[d] = d3.scale.linear()
```
Comment (callicles) [inline public/javascripts/repository.js]: Codacy Issue found: [Expected '!==' and instead saw '!='.](https://www.codacy.com/app/gitlinks/gitrank-web/file/993762770/issues/source?bid=2432310&fileBranchId=2868924#l63)

File: public/javascripts/repository.js (hunk 6 above)
Comment (callicles) [inline public/javascripts/repository.js]: Codacy Issue found: [Variable 'dimensions' implicitly declared.](https://www.codacy.com/app/gitlinks/gitrank-web/file/996075795/issues/source?bid=2432310&fileBranchId=2868924#l62)

File: public/javascripts/repository.js (hunk 7 above)
Comment (callicles) [inline public/javascripts/repository.js]: Codacy Issue found: [Expected '!==' and instead saw '!='.](https://www.codacy.com/app/gitlinks/gitrank-web/file/996075795/issues/source?bid=2432310&fileBranchId=2868924#l63)

File: app/models/services/RepositoryService.scala (hunk 2 above)
Comment (jacobtardieu) [inline app/models/services/RepositoryService.scala]: Yes but in that case you should not use it (cf what I wrote above).

File: public/javascripts/repository.js (hunk 6 above)
Comment (callicles) [inline public/javascripts/repository.js]: Corrected

File: conf/routes (hunk 5 above)
Comment (callicles) [inline conf/routes]: Corrected

File: app/models/services/RepositoryService.scala [hunk 8]
```diff
@@ -162,31 +142,26 @@ class RepositoryService @Inject()(
...
    */
-  def getPermissionToFeedback(repoName: String, user: Option[User]): Future[Boolean] = {
-    user match {
-      case Some(userEntity) =>
-        contributionDAO.checkIfUserContributed(userEntity.username, repoName).map {
-          hasContributed => !hasContributed
-        }
-      case None => Future(false)
-    }
-  }
```
Comment (brunnoattorre) [inline app/models/services/RepositoryService.scala]: I think you shouldn't have deleted this method. There are two kinds of permission that need to be checked:

1- If the user can give any kind of feedback (for instance, if you contributed to repo A you can't give feedback to it).
2- If 1 is true, then we have to check if the user can add a feedback or just updated (based on if he had scored it or not).

File: app/controllers/ApplicationController.scala [hunk 9]
```diff
@@ -56,7 +56,7 @@ class ApplicationController @Inject()(
     repoService.getFromNeoOrGitHub(request.identity, repoName).flatMap({
       case Some(repository) => repoService.getFeedback(repoName, page).flatMap((feedback: Seq[Feedback]) =>
         repoService.getFeedbackPageCount(repoName).flatMap(totalPage => {
-          repoService.getPermissionToFeedback(repoName, request.identity).map (
+          repoService.getPermissionToAddFeedback(repoName, request.identity).map (
```
Comment (brunnoattorre) [inline app/controllers/ApplicationController.scala]: I think you shouldn't have deleted this method. There are two kinds of permission that need to be checked:

1- If the user can give any kind of feedback (for instance, if you contributed to repo A you can't give feedback to it).
2- If 1 is true, then we have to check if the user can add a feedback or just updated (based on if he had scored it or not).

That's why there were 2 methods.

File: app/controllers/ApplicationController.scala [hunk 10]
```diff
@@ -75,7 +75,7 @@ class ApplicationController @Inject()(
    */
   def giveFeedbackPage(owner: String, repositoryName: String) = UserAwareAction.async { implicit request =>
     val repoName: String = owner + "/" + repositoryName
-    repoService.getPermissionToFeedback(repoName, request.identity).flatMap {
+    repoService.getPermissionToAddFeedback(repoName, request.identity).flatMap {
```
Comment (brunnoattorre) [inline app/controllers/ApplicationController.scala]: I think you shouldn't have deleted this method. There are two kinds of permission that need to be checked:

1- If the user can give any kind of feedback (for instance, if you contributed to repo A you can't give feedback to it).
2- If 1 is true, then we have to check if the user can add a feedback or just updated (based on if he had scored it or not).

That's why there were 2 methods.

**Pull request merged**